File: CPP/Queue/Queue.cpp
#include <iostream>
#define max 10


using namespace std;

class queue
{
    private:
        int Front,rear;
        int arr[max];
    public:
        queue()
        {Front=rear=-1;}
        void enqueue(int);
        int dequeue();
        void display();
};

void queue::enqueue(int item)
{
    if(rear==max-1)
        cout<<"Queue Full!"<<endl;
    else
    {
        rear++;
        arr[rear]=item;
        if(Front==-1)
            Front=0;

    }
}
int queue::dequeue()
{
    int data;
    if(Front==-1)
        return -2;
    else
    {
        data=arr[Front];
        arr[Front]=0;
        if(rear==Front)
            Front=rear=-1;
        else
            Front++;

        return data;
    }


}

void queue::display()
{
    if(Front==-1)
    cout<<"Queue Empty!"<<endl;
    else
    {
        for(int i=Front;i<=rear;i++)
            cout<<arr[i]<<endl;
    }

}
int main()
{

    queue q1;
    int item,choice;

    while(choice!=4)
    {
        cout<<"Enter your choice"<<endl;
        cout<<"1. Enqueue"<<endl<<"2. Dequeue"<<endl<<"3. Display"<<endl<<"4. Exit"<<endl;
        cin>>choice;
        switch(choice)
        {
            case 1:cout<<"Enter the element to be enqueued"<<endl;
                    cin>>item;
                    q1.enqueue(item);

                    break;
            case 2: item=q1.dequeue();
                    if(item==-2)
                        cout<<"Queue Empty!"<<endl;
                    else
                    {
                        cout<<"Element dequeued is"<<endl;
                        cout<<item<<endl;
                    }

                    break;
            case 3:cout<<"The elements of the queue are:"<<endl;
                    q1.display();
                    break;
            case 4:choice=4;
                    break;
            default:cout<<"Invalid choice"<<endl;
        }
    }

    return 0;
}

File: Java/Queue/Queue.java
import java.util.Scanner;

public class Queue {
    private final int QUEUE_SIZE = 10;
    private int front;
    private int rear;
    private int q[];

    Queue() {
        front = rear = -1;
        q = new int[QUEUE_SIZE];
    }

    public int getMaxSize() {
        return QUEUE_SIZE;
    }

    public boolean isEmpty() {
        if (front == -1 && rear == -1) {
            return true;
        }
        return false;
    }

    public void enqueue(int val) {
        if ((rear + 1) % QUEUE_SIZE == front) {
            System.out.println("Queue is full");
            return;
        }
        if (isEmpty()) {
            front = rear = 0;
        } else {
            rear = (rear + 1) % QUEUE_SIZE;
        }
        q[rear] = val;
    }

    public void dequeue() {
        if (isEmpty()) {
            System.out.println("Queue is empty");
            return;
        } else if (front == rear) {
            front = rear = -1;
        } else {
            front = (front + 1) % QUEUE_SIZE;
        }
    }

    public int peek() {
        if (isEmpty()) {
            System.out.println("Queue is empty");
            return Integer.MIN_VALUE;
        }
        return q[front];
    }

    public int getSize() {
        if (front > rear) {
            return QUEUE_SIZE - (front - rear);
        } else {
            return rear - front + 1;
        }
    }

    public static void main(String args[]) {
        Queue q = new Queue();
        Scanner sc = new Scanner(System.in);
        System.out.println("Enter the number of elements: ");
        int n = sc.nextInt();
        if (n > q.getMaxSize()) {
            System.out.println("Input size is more than the maximum queue size");
            System.exit(-1);
        }
        for (int i = 0; i < n; i++) {
            q.enqueue(sc.nextInt());
        }
        System.out.println("Queue size: " + q.getSize());
        System.out.println("Queue elements: ");
        while (!q.isEmpty()) {
            System.out.print(q.peek() + " ");
            q.dequeue();
        }
        System.out.println();
    }
}
